Source/WebCore/page/scrolling/ScrollingCoordinatorTypes.cpp

```diff
@@ -60,9 +60,17 @@ void RequestedScrollData::merge(RequestedScrollData&& other)
     *this = WTFMove(other);
 }
 
-FloatPoint RequestedScrollData::destinationPosition(const FloatPoint& currentScrollPosition) const
+FloatPoint RequestedScrollData::destinationPosition(FloatPoint currentScrollPosition) const
 {
-    return requestType == ScrollRequestType::DeltaUpdate ? (currentScrollPosition + std::get<FloatSize>(scrollPositionOrDelta)) : std::get<FloatPoint>(scrollPositionOrDelta);
+    return computeDestinationPosition(currentScrollPosition, requestType, scrollPositionOrDelta);
+}
+
+FloatPoint RequestedScrollData::computeDestinationPosition(FloatPoint currentScrollPosition, ScrollRequestType requestType, const std::variant<FloatPoint, FloatSize>& scrollPositionOrDelta)
+{
+    if (requestType == ScrollRequestType::DeltaUpdate)
+        return currentScrollPosition + std::get<FloatSize>(scrollPositionOrDelta);
```

Comment:
it would be nice if there was somehow a way to codify the invariant that `ScrollRequestType::DeltaUpdate` corresponds to `scrollPositionOrDelta` being a `FloatSize` and ditto for `FloatPoint`, but this is unrelated to this specific PR.